include/libtorrent/session_handle.hpp

```diff
@@ -443,6 +443,19 @@ namespace libtorrent
 		void dht_get_peers(sha1_hash const& info_hash);
 		void dht_announce(sha1_hash const& info_hash, int port = 0, int flags = 0);
 
+		// Iterate all the live DHT (identified by ``nid``) nodes and pass the
+		// nodes id and endpoint to the callback ``cb``. If the callback function
+		// returns true, the pair will be returned in the list of nodes in the
+		// alert ``dht_live_nodes_alert``.
+		// Since this alert is a response to an explicit call, it will always be
+		// posted, regardless of the alert mask.
+		//
+		// Given that the function is called from an internal thread of libtorrent,
```

Comment:
this kind of API makes me nervous. do you think there's a simpler way of specifying filters (without a callback) that would cover the majority of use cases?